src/plugins/hiddenColumns/contextMenuItem/showColumn.js

```diff
@@ -38,13 +38,9 @@ export default function showColumnItem(hiddenColumnsPlugin) {
 
       const startVisualColumnAfterAction = this.toVisualColumn(startPhysicalColumn);
       const endVisualColumnAfterAction = this.toVisualColumn(endPhysicalColumn);
-      const allColumnsSelected = endVisualColumnAfterAction - startVisualColumnAfterAction + 1 === this.countCols();
-      // TODO: Workaround, because selection doesn't select headers properly in a case when we select all columns
-      // from `0` to `n`, where `n` is number of columns in the `DataMap`.
-      const selectionStart = allColumnsSelected ? -1 : startVisualColumnAfterAction;
 
```

Comment:
I removed that part of code because it had wrong assumptions? The `selectionStart` in some cases had -1 value which was passed to `hot.selectColumns()`. But when I read the `selectColumns` method implementation it occurs that the method doesn't support negative values. I didn't notice any wrong behavior after removing those lines.

Reply:
Please look at [this comment](https://github.com/handsontable/handsontable/pull/6887#discussion_r427251144).

Reply:
Probably, If I remember correctly reverting that line didn't have any effect on that test which you mentioned. But to be sure I will check again.

Reply:
I've checked once again and yeah, I was wrong the code works. I reverted the changes and applied the same fix for the _HiddenRows_ plugin https://github.com/handsontable/handsontable/pull/6887/commits/18642c892a50958e0bf3fffb23c9d758902d6d52.